ipn/prefs.go

```diff
@@ -616,7 +627,8 @@ func (p *Prefs) Equals(p2 *Prefs) bool {
 		p.AppConnector == p2.AppConnector &&
 		p.PostureChecking == p2.PostureChecking &&
 		slices.EqualFunc(p.DriveShares, p2.DriveShares, drive.SharesEqual) &&
-		p.NetfilterKind == p2.NetfilterKind
+		p.NetfilterKind == p2.NetfilterKind &&
+		reflect.DeepEqual(p.RelayServerPort, p2.RelayServerPort)
```

Comment:
kinda overkill ... and looks like this code is already trying to avoid it

Reply:
Overkill in what sense? Importing reflect vs writing a similar equality check by hand?

Reply:
Updated in https://github.com/tailscale/tailscale/pull/15594/commits/68a710250304e113561db2b36e733a179b7e2fb3.